packages/zoe/src/contracts/coveredCall-durable.js

```diff
@@ -0,0 +1,151 @@
+ * itself: an inspectable invitation to buy the underlying assets. The
+ * call option invitation has this additional information in the
+ * value: {expirationDate, timeAuthority, underlyingAssets,
+ * strikePrice }
+ *
+ * The invitation itself can be traded as a valuable digital asset: a
+ * covered call option.
+ *
+ * The recipient of a covered call option (buying it on some other
+ * exchange or through some other trading contract) can exercise the
+ * option before the deadline by using the option as an invitation to
+ * this contract, paying the strike price and receiving the underlying
+ * assets. The recipient of a covered call option can use whatever
+ * keywords they wish, as long as they give the strike price as
+ * specified in the invitation value, and want the underlying assets
+ * exactly.
+ *
+ * @param {import('@agoric/vat-data').Baggage} installationBaggage
+ * @param {ZCF} zcf passed in setupInstallation, but not completely initialized
+ *    until setupInstance. Passed early because it will often need to be
+ *    captured lexically by durable kinds.
+ */
+const setupInstallation = (installationBaggage, zcf) => {
+  const instanceBaggage = provideDurableMapStore(
+    installationBaggage,
+    'instance',
+  );
+
+  const makeExerciserKindHandle = provideKindHandle(
+    instanceBaggage,
+    'makeExerciserKindHandle',
+  );
+
+  const makeExerciser = defineDurableKind(
+    makeExerciserKindHandle,
+    sellSeat => ({ sellSeat }),
+    {
+      handle: ({ state: { sellSeat } }, buySeat) => {
+        assert(!sellSeat.hasExited(), sellSeatExpiredMsg);
+        try {
+          swapExact(zcf, sellSeat, buySeat);
+        } catch (err) {
+          console.log(
+            'Swap failed. Please make sure your offer has the same underlyingAssets and strikePrice as specified in the invitation details. The keywords should not matter.',
+          );
+          throw err;
+        }
+        zcf.shutdown('Swap completed.');
+        return `The option was exercised. Please collect the assets in your payout.`;
+      },
+    },
+  );
+
+  /** @type {OfferHandler} */
+  const makeOption = sellSeat => {
+    fit(sellSeat.getProposal(), M.split({ exit: { afterDeadline: M.any() } }));
+    const sellSeatExitRule = sellSeat.getProposal().exit;
+    assert(
+      isAfterDeadlineExitRule(sellSeatExitRule),
+      `the seller must have an afterDeadline exitRule, but instead had ${sellSeatExitRule}`,
+    );
+
+    const exerciseOption = makeExerciser(sellSeat);
+    const customProps = harden({
+      expirationDate: sellSeatExitRule.afterDeadline.deadline,
+      timeAuthority: sellSeatExitRule.afterDeadline.timer,
+      underlyingAssets: sellSeat.getProposal().give,
+      strikePrice: sellSeat.getProposal().want,
+    });
+    // @ts-expect-error durable handlers need typing
+    return zcf.makeInvitation(exerciseOption, 'exerciseOption', customProps);
+  };
+
+  const setupInstance = _privateArgs => {
```

Comment:
this `_privateArgs` is serving to document the API. let's leave that to JSDoc

Reply:
I feel much less strongly about this one, but I do often like to put these names in the JS itself to remind a reader what's missing. YMMV.